probe-rs/src/probe/jlink/mod.rs

```diff
@@ -459,15 +459,22 @@ impl DebugProbe for JLink {
 
         log::debug!("Attaching with protocol '{}'", actual_protocol);
 
+        let jlink = self.handle.get_mut().unwrap();
+
+        // Verify target voltage (VTref pin, mV). If this is 0, the device is not powered.
+        let target_voltage = jlink.read_target_voltage()?;
+        if target_voltage == 0 {
```

Comment:
I think we should compare this against a threshold voltage because I am not sure the JLink returns exactly 0. Or did you test this? Maybe < 0.1 would be more suitable?

Reply:
I tested it, and it seems to work. I assume it's measured using a pull down (otherwise with a floating input you'd get random values).

Reply:
> I tested it, and it seems to work. I assume it's measured using a pull down (otherwise with a floating input you'd get random values).

Ok, if it's well tested that's good enough for me :) But having a pull down might not be enough with noise, depending on the ADC ;) Hence my question :)

Reply:
Yes, that is true. If I ever notice that a low-but-not-0 voltage is measured, I'll update the code :slightly_smiling_face: